src/transformers/models/whisper/tokenization_whisper_fast.py

```diff
@@ -485,3 +490,30 @@ def _decode_asr(self, model_outputs, *, return_timestamps, return_language, time
             return_language=return_language,
             time_precision=time_precision,
         )
+
+    # Copied from transformers.models.whisper.tokenization_whisper.WhisperTokenizer.get_prompt_ids
+    def get_prompt_ids(self, text: str, return_tensors="np"):
+        """Converts prompt text to IDs that can be passed to [`~WhisperForConditionalGeneration.generate`]."""
+        batch_encoding = self("<|startofprev|>", text.strip(), add_prefix_space=True, add_special_tokens=False)
```

Comment:
Quick comment here. By default, `WhisperTokenizerFast` sets `add_prefix_space` to `False`. After being inititalized, the value cannot be changed (see related #17391).

```python
from transformers import WhisperTokenizerFast           
                                                                      
tokenizer = WhisperTokenizerFast.from_pretrained("openai/whisper-tiny")
                                                                      
tokenizer("<|startofprev|>", "test", add_special_tokens=False).input_ids
> [50361, 31636]

tokenizer("<|startofprev|>", "test", add_special_tokens=False, add_prefix_space=True).input_ids   
> TypeError: PreTrainedTokenizerFast._batch_encode_plus() got an unexpected keyword argument 'add_prefix_space'
```

Is it necessary to have `add_prefix_space` be `False` by default? Pinging @hollance and @sanchit-gandhi too.

Reply:
Since Whisper always infers an extra space after the `<|startoftranscript|>` / `<|startofprev|>`, I see why it could make sense to change `add_prefix_space` to `True` by default. Probably we should have set it to `True` when we added the model in the first place, but right now to swap it would be a breaking change since we would silently change the behaviour of the tokenizer

I'm not sure there's a clean way we can do this directly? Probably easiest is to instantiate the tokenizer with `add_prefix_space=True` where required:
```python
from transformers import WhisperTokenizerFast

tokenizer = WhisperTokenizerFast.from_pretrained("openai/whisper-tiny", add_prefix_space=True)
```

Reply:
For training with initial prompts, right now we are using two tokenizers, which is not ideal.